onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -31,28 +31,28 @@ class SVMCommon {
 
   float kernel_dot(const T* A, int64_t a, const std::vector<float>& B, int64_t b, int64_t len, KERNEL k) const {
     float sum = 0.f;
```

Comment:
This function will only work when T=`float` with your changes (because of `++pA` and `++pB` in for-loops). Is this desired?

Reply:
Fixed. There is no test for double. Should I remove the template and only allow floats?